src/SprykerEco/Zed/AfterPay/Communication/Plugin/Checkout/AfterPayPostCheckPlugin.php

```diff
@@ -34,6 +35,10 @@ class AfterPayPostCheckPlugin extends AbstractPlugin implements CheckoutPostSave
      */
     public function executeHook(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse): void
     {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== SharedAfterPayConfig::PROVIDER_NAME) {
```

Comment:
* Plugin doesn't contain business logic and just delegate to the underlying Module API (Facade, Client, or Service).
* Move the condition under the hood of the `postSaveHook()` method.